subsys/bluetooth/controller/hci/hci.c

```diff
@@ -22,6 +22,8 @@
 #include <bluetooth/buf.h>
 #include <bluetooth/bluetooth.h>
 
+#include "../host/hci_ecc.h"
```

Comment:
This seems incompatible with a dual core architecture. Does the `#if defined` wrapper on line 756 save you from compilation errors on nRF53? Would it be more appropriate to place this in a static inline or a define in a header in subsys/bluetooth/common instead, since the feature set is determined by the same logic?

Reply:
The relative path you mean? It compiles fine.

This wasn't pretty in any way. But I couldn't find a way to make this elegant without larger changes.
CONFIG_BT_TINYCRYPT_ECC is selectable in a controller only build (which actually surprised me), ideally I would have moved this part to hci_raw.c.

Reply:
I was thinking more about calling functions in a host module from the controller, but if hci_ecc.c is built in controller-only configurations, I suppose it should compile. AFAICT, you're only using the new supported_features function, which is why I'm suggesting to make that a static inline function under bluetooth/common

Reply:
In a controller only build with CONFIG_BT_TINYCRYPT_ECC=y then hci_raw.c and hci_ecc.c are both included, these both are in the host folder.
All of the functions in hci_ecc.h would be used in this case, but all the other ones are called from hci_raw.c, which is why I'd ideally would have moved this handling to hci_raw.c as well.

I think the cleanest would be to move BT_TINYCRYPT_ECC to bluetooth/common/kconfig and move hci_ecc.c/h to bluetooth/common.

Reply:
That is what I'm proposing, really. But it doesn't block the changes made in this PR, so I won't hold it up any more.